### PR TITLE
Enable priorities to be fetched from file

### PR DESCRIPTION
**Pull request by sidneymbell**
## Description of proposed changes

Currently, there's not a great way to specify a `priorities.tsv` file from within the `builds.yaml`. This small PR enables one to specify a priorities file like so:

```
## within builds.yaml
subsampling:
  test_subsampling_schema:
    test_subsampling_schema_tier:
      group_by: "country"
      max_sequences: 5
      priorities:
        type: "file"
        file: "my_profiles/example/priorities.tsv"
```

This is useful for many situations. For example, I'm currently trying to define a build that prioritizes samples based on a combination of their collection date and their submission date, which isn't currently possible without being able to load in a `priorities.tsv` for one particular `subsampling_schema_tier`. 

## Related issue(s)
I ended up needing this one on the fly and thought it would be useful upstream for other folks. I don't see a current issue open for it, though. Would love any and all feedback you have! 

Fixes #
Related to #

## Testing

What steps should be taken to test the changes you've proposed? 
**NOTE: I included two demo files here to make testing easy, but will remove them before merge.**
Ran builds using a demo dataset, configured as above. Also ran a vanilla `getting_started` build (w/o a priorities file). Both worked as expected :)

If you added or changed behavior in the codebase, did you update the tests, or do you need help with this?

## Release checklist

If this pull request introduces new features, complete the following steps:

 - [x] Update `docs/change_log.md` in this pull request to document these changes by the date they were added.

### CODE REVIEW

**Review by huddlej — CHANGES_REQUESTED** [review]
This is a great addition, @sidneymbell! I tried it out with the "getting started" profile, but when subsampling groups by country with 5 max sequences, it's hard to tell whether the selected sequences are the higher priority strains. For example, in my test run I got the following strains from subsampling:

```
ENV/USA/UF-11/2020
India/GBRC56a/2020
Japan/P4-6/2020
USA/VA-DCLS-0077/2020
USA/WA-UW-1338/2020
Wuhan/Hu-1/2019
Wuhan/WH01/2019
```

It seems like Australia got randomly dropped in this run. I tried removing the group constraint from the subsampling scheme such that the 5 selected sequences should be those with the highest priority in your given file. With this change, I got the following strains:

```
Australia/VIC300/2020
Australia/VIC302/2020
Wuhan/Hu-1/2019
Wuhan/WH01/2019
```

The two Wuhan reference strains got added back because they were in the `include.txt` file. The two Australian strains have the highest priorities in the given file (36 and 37, respectively). Is this what you would have expected for the example priorities file? Or were you anticipating that `Australia/VIC05/2020` and `Australia/VIC1000/2020` would have made the list instead? (I had to check this again for myself to confirm that higher numbers in the file indicate higher priorities.)

As part of this PR, could you also update [the corresponding section about subsampling priorities in the configuration guide](https://nextstrain.github.io/ncov/configuration.html#priorities) (source is in `docs/configuration.md`) and the change log (`docs/change_log.md`) to mention this new option? For the change log, we've been using a new header like `## New features since last version update` above the last major release to note new features.

**Comment by sidneymbell** [comment]
Thanks for the review, @huddlej! 

Apologies, I'd forgotten that I made myself a tiny demo dataset that only had sequences from Australia. In short, yes, this is the expected behavior! I had to double check the low- vs high-priorities myself :)

I'll go ahead and add to the docs and change log. Any other changes blocking merge?

**Review by huddlej — APPROVED** [review]
Looks great, @sidneymbell! Thank you!

I just resolved the conflict with `master`. I'll merge, once CI passes.

**Comment by sidneymbell** [comment]
Awesome, thanks!

**Pull request merged**
